### ISSUE TITLE
Jobs skipped during DST change

### ISSUE THREAD

**Issue by leshik** [rundeck/rundeck#6879]
Can't reopen https://github.com/rundeck/rundeck/issues/3234 thus creating a new issue. Had a very critical job skipped yesterday due to this. I understand that it's not Rundeck's fault that 2:00 am – 2:59 am Eastern Time technically hadn't existed yesterday, thus not sure if it's a bug or feature request. I just think Rundeck could do better to mitigate this class of issues.

**Comment by Khelben-Arunsun** [comment]
2:00 am – 2:59 Does not exist. Only an offset to the fold does. 
Don't set your time to DST and you will not have that issue. 
This is more a political matter and NOT a software problem 

You have to Fold Time in order to see the issue. 
From the past looking into the future you see the 0 side.
From the future looking back you see the 1 side. 
& you have to be looking from UTC time and not from a daylight savings time view.
Here is an example with a Z offset.  
`

# time viewed across the daylight savings time fallbackThe CST attribute for 1 AM at 06:UTC shows the folded attribute of one. 04:00:00 UTC = 00:00:00 CDT 005:00:00 UTC = 01:00:00 CDT 006:00:00 UTC = 01:00:00 CST 107:00:00 UTC = 02:00:00 CST 0  # In numpy this is shown with a Z as a Z offset attribute.datetime_as_string(d, timezone='UTC')array(['2002-10-27T04:30Z', '2002-10-27T05:30Z', '2002-10-27T06:30Z',       '2002-10-27T07:30Z'], dtype='<U35')
--


`

**Comment by Khelben-Arunsun** [comment]
https://www.quartz-scheduler.org/documentation/2.3.1-SNAPSHOT/faq.html


If you read quartz scheduling docs. 

CronTrigger allows you to schedule jobs to fire at certain moments with respect to a "Gregorian calendar". Hence, if you create a trigger to fire every day at 10:00 am, before and after daylight savings time switches it will continue to do so. However, depending on whether it was the Spring or Autumn daylight savings event, for that particular Sunday, the actual time interval between the firing of the trigger on Sundary morning at 10:00 am since its firing on Saturday morning at 10:00 am will not be 24 hours, but will instead be 23 or 25 hours respectively.



## There is one additional point users must understand about CronTrigger with respect to daylight savings. 
### This is that you should take careful thought about creating schedules that fire between midnight and 3:00 am (the critical window of time depends on your trigger’s locale, as explained above). The reason is that depending on your trigger’s schedule, and the particular daylight event, the trigger may be skipped or may appear to not fire for an hour or two. As examples, say you are in the United States, where daylight savings events occur at 2:00 am. If you have a CronTrrigger that fires every day at 2:15 am, then on the day of the beginning of daylight savings time the trigger will be skipped, since, 2:15 am never occurs that day. If you have a CronTrigger that fires every 15 minutes of every hour of every day, then on the day daylight savings time ends you will have an hour of time for which no triggerings occur, because when 2:00 am arrives, it will become 1:00 am again, however all of the firings during the one o’clock hour have already occurred, and the trigger’s next fire time was set to 2:00 am - hence for the next hour no triggerings will occur.

In summary, all of this makes perfect sense, and should be easy to remember if you keep these two rules in mind:

**Comment by fdevans** [comment]
Thank you @Khelben-Arunsun.  Rundeck does use Quartz and that advice is valid.  Since TimeZones and DST changes vary widely across the world and business processes vary with each company using Rundeck it would impossible for Rundeck to make the "right" choice of when to schedule if the clock didn't match the schedule.

Rundeck Enterprise does provide very flexible scheduling options:
- Repeated Schedules combined with the Calendaring option could be used to run the job on an alternative schedule for the days when the time changes.

- In addition the [Missed Job Fires](https://docs.rundeck.com/docs/manual/schedules/missedjobfires.html#missed-job-fires-enterprise-only) feature might also help here.  I didn't have time to test specifically a DST type of change for a missed schedule, but someone on the team is going to check it out.

Overall the current offering is a very comprehensive and flexible Scheduling and Calendaring solution that would provide the ability to implement the schedule that fits for each individual business's needs.  If there is a specific enhancement that might help please share in our [Google Forums](https://groups.google.com/g/rundeck-discuss?pli=1).
